pkg/util/identifiable/identifiable_test.go

```diff
@@ -0,0 +1,109 @@
+package identifiable
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type someId struct {
+	// name is a "key" field used for merging
+	name string
+	// some other property. Indicates which exactly object was returned by an aggregation operation
+	property string
+}
+
+func newSome(name, property string) someId {
+	return someId{
+		name:     name,
+		property: property,
+	}
+}
+
+func (s someId) Identifier() interface{} {
+	return s.name
+}
+
+func Test_SetDifference(t *testing.T) {
+	oneLeft := newSome("1", "left")
+	twoLeft := newSome("2", "left")
+	twoRight := newSome("2", "right")
+	threeRight := newSome("3", "right")
+	fourRight := newSome("4", "right")
+
+	left := []Identifiable{oneLeft, twoLeft}
+	right := []Identifiable{twoRight, threeRight}
+
+	assert.Equal(t, []Identifiable{oneLeft}, SetDifference(left, right))
+	assert.Equal(t, []Identifiable{threeRight}, SetDifference(right, left))
+
+	left = []Identifiable{oneLeft, twoLeft}
+	right = []Identifiable{threeRight, fourRight}
+	assert.Equal(t, left, SetDifference(left, right))
+
+	left = []Identifiable{}
+	right = []Identifiable{threeRight, fourRight}
+	assert.Empty(t, SetDifference(left, right))
+	assert.Equal(t, right, SetDifference(right, left))
+
+	left = nil
+	right = []Identifiable{threeRight, fourRight}
+	assert.Empty(t, SetDifference(left, right))
+	assert.Equal(t, right, SetDifference(right, left))
```

Comment:
yeah good catch, I tried them a couple of times generated by Idea and the code was more verbose than just sequential checks. 
Maybe it's just Idea generating too heavy structures.
Will try 👍

Reply:
yes, idea usually goes to heavy on the testcase definition I think this can be simple enough to be something like
```go
test_cases := map[string]struct {
	left []Identifiable
	right []Identifiable
}{
	"name": {
		left: //values
		right: //values
	},
}
for name, tc := range testCases {
	t.Run(name, func(t *testing.T) {
		assert.Empty(t, SetDifference(tc.left, tc.right)) // go lint may flag this about functions in loops and pointers
 		assert.Equal(t, right, SetDifference(tc.right, tc.left))
	})
}
```